tests/filters/parser_filter.py

```diff
@@ -0,0 +1,169 @@
+        {'excluded': set(['plugin1', 'plugin2'])}, {'excluded': set(['*'])})
+    self.assertEqual(expression, '!excluded/plugin1,!excluded/plugin2,excluded')
+
+    expression = test_helper._JoinExpression({}, {'included': set(['*'])})
+    self.assertEqual(expression, 'included')
+
+    expression = test_helper._JoinExpression(
+        {}, {'included': set(['*', 'plugin1'])})
+    self.assertEqual(expression, 'included,included/plugin1')
+
+    expression = test_helper._JoinExpression(
+        {}, {'included': set(['plugin1', 'plugin2'])})
+    self.assertEqual(expression, 'included/plugin1,included/plugin2')
+
+    expression = test_helper._JoinExpression(
+        {'excluded': set(['plugin1'])},
+        {'excluded': set(['*']), 'included': set(['plugin2'])})
+    self.assertEqual(expression, '!excluded/plugin1,excluded,included/plugin2')
+
+    with self.assertRaises(RuntimeError):
+      test_helper._JoinExpression(
+          {'excluded': set(['plugin1'])}, {'excluded': set(['plugin1'])})
+
+    with self.assertRaises(RuntimeError):
+      test_helper._JoinExpression({'excluded': set(['plugin1'])}, {})
+
+  def testExpandPresets(self):
+    """Tests the ExpandPresets function."""
+    presets_file = self._GetTestFilePath(['presets.yaml'])
+    self._SkipIfPathNotExists(presets_file)
+
+    presets_manager = parsers_presets.ParserPresetsManager()
+    presets_manager.ReadFromFile(presets_file)
+
+    test_helper = parser_filter.ParserFilterExpressionHelper(presets_manager)
+
+    expected_parser_filter_expression = ','.join(sorted([
+        '!utmp',
+        'bencode',
+        'binary_cookies',
+        'chrome_cache',
+        'chrome_preferences',
+        'czip/oxml',
+        'esedb',
+        'esedb/msie_webcache',
+        'filestat',
+        'firefox_cache',
+        'gdrive_synclog',
+        'java_idx',
+        'lnk',
+        'mcafee_protection',
+        'msiecf',
+        'olecf',
+        'opera_global',
+        'opera_typed_history',
+        'pe',
+        'plist/safari_history',
+        'prefetch',
+        'sccm',
+        'skydrive_log',
+        'skydrive_log_old',
+        'sqlite/chrome_27_history',
+        'sqlite/chrome_8_history',
+        'sqlite/chrome_autofill',
+        'sqlite/chrome_cookies',
+        'sqlite/chrome_extension_activity',
+        'sqlite/firefox_cookies',
+        'sqlite/firefox_downloads',
+        'sqlite/firefox_history',
+        'sqlite/google_drive',
+        'sqlite/skype',
+        'symantec_scanlog',
+        'usnjrnl',
+        'winfirewall',
+        'winjob',
+        'winreg']))
+
+    parser_filter_expression = test_helper.ExpandPresets('win_gen,!utmp')
+    self.assertEqual(
+        parser_filter_expression, expected_parser_filter_expression)
+
+    parser_filter_expression = test_helper.ExpandPresets('olecf,!utmp')
+    self.assertEqual(parser_filter_expression, '!utmp,olecf')
+
+  def testSplitExpression(self):
+    """Tests the SplitExpression function."""
```

Comment:
Please add a test here where a preset name is specified in the parser filter expression.

Reply:
what value does that add? a preset is treated like any other parser definition by this method?

Reply:
This is clearer now, but I still wonder in the future about this method changing to expand presets.

Reply:
> This is clearer now, but I still wonder in the future about this method changing to expand presets.

Can you be more specific, what do you wonder about. What is the concern? How do you think it should be addressed?